immunopepper/preprocess.py

```diff
@@ -362,189 +362,198 @@ def parse_gene_metadata_info(h5fname, sample_list, cross_graph_expr):
-    if file_type == 'h5': # hdf5 filr
-        mutation_dic = parse_mutation_from_vcf_h5(vcf_path, target_sample_list, mutation_sample, heter_code, sample_eq_dict)
-        logging.info("Get germline mutation dict from h5 file in {}. No pickle file created".format(vcf_path))
-        return mutation_dic
-    else: # vcf text file
-        f = open(vcf_path,'r')
-        lines = f.readlines()
-        mutation_dic = {}
-        for line in lines:
-            if line.strip()[:2] == '##':  # annotation line
-                continue
-            if line.strip()[0] == '#':  # head line
-                fields = line.strip().split('\t')
-                file_sample_list = fields[9:]
-                for target_sample in target_sample_list:
-                    if (sample_eq_dict[target_sample] not in file_sample_list) and (mutation_sample == target_sample):
-                        logging.error("samples in mutation file: {}".format(file_sample_list))
-                        logging.error("No mutations for sample {} found in vcf file, please check samples above, consider using --sample-name-map ".format(sample_eq_dict[target_sample]))
-                        sys.exit(1)
-                continue
-            items = line.strip().split('\t')
-            var_dict = {}
-            chr = items[0]
-            pos= int(items[1])-1
-            var_dict['ref_base'] = items[3]
-            var_dict['mut_base'] = items[4]
-            var_dict['qual'] = items[5]
-            var_dict['filter'] = items[6]
-            if len(var_dict['ref_base']) == len(var_dict['mut_base']):  # only consider snp for now
-                for i, file_sample in enumerate(file_sample_list):
-                    if items[9+i].split(':')[0] in {'1|1' ,'1|0', '0|1', '0/1', '1/0', '1/1'}:
-                        if file_sample not in sample_eq_dict_reverse.keys():
-                            sample_eq_dict_reverse[file_sample] = file_sample
-                        if (sample_eq_dict_reverse[file_sample] ,chr) in list(mutation_dic.keys()):
-                            mutation_dic[(sample_eq_dict_reverse[file_sample],chr)][int(pos)] = var_dict
-                        else:
-                            mutation_dic[(sample_eq_dict_reverse[file_sample],chr)] = {}
-                            mutation_dic[(sample_eq_dict_reverse[file_sample], chr)][int(pos)] = var_dict
-    if mut_pickle:
-        f_pkl =open(vcf_pkl_file,'wb')
-        pickle.dump(mutation_dic,f_pkl)
-        logging.info("create vcf pickled mutation dict for next time's use in {}".format(vcf_pkl_file))
-
-    return mutation_dic
-
-
-def parse_mutation_from_vcf_h5(h5_vcf_path, target_sample_list, mutation_sample, heter_code=0, sample_eq_dict={}):
-    """
-    Extract germline mutation information from given vcf h5py file.
+    if file_type == 'h5':  # hdf5 file
+        mutation_dict = parse_mutation_from_vcf_h5(vcf_path, mutation_sample, heter_code)
+        # TODO(reviewers): why no pickle created?
```

Comment:
Yes, why not?

Reply:
I don't know why this choice was made. If it is much faster than preprocessing the hdf5 again, then I am fine with a pickle.